### ISSUE TITLE
Emacs Magit doesn’t work on 23.11

### ISSUE THREAD

**Issue by Nebucatnetzer** [NixOS/nixpkgs#272019]
### Describe the bug
When on tries to open magit on 23.11 you get the `^L` in the output and the commands aren’t working as expected.

### Steps To Reproduce
1. Update to 23.11
2. Start Emacs and Magit in a Git repo of your choice.
3. Try to select a commit for example.

### Expected behavior
The commit diff should be shown.

### Screenshots
![image](https://github.com/NixOS/nixpkgs/assets/2287221/432dc0b0-c499-4c56-a1d8-b6ddaae454ac)

### Additional context
The issue is already fixed upstream but requires some changes in Nixpkgs:
- https://github.com/magit/magit/issues/5011
- https://github.com/magit/magit/issues/5011#issuecomment-1836861539

### Notify maintainers

@AndersonTorres
@adisbladis
@Atemu 
@jwiegley
@lovek323
@matthewbauer

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.1.64, NixOS, 23.11 (Tapir), 23.11.20231202.933d7dc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - nixpkgs: `/nix/store/chi01x4fh8hx7wwh0rqd7m1c8583iyhv-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by AndersonTorres** [comment]
I reproduced the error here - with the same seq-keep error message.

**Comment by Nebucatnetzer** [comment]
This is still broken.

**Comment by AndersonTorres** [comment]
It would require updating elisp-packages on stable.

**Comment by Nebucatnetzer** [comment]
I know, isn’t that approapriate when such an important package is broken?

**Comment by AndersonTorres** [comment]
It became a bit too hard recently, since Emacs triggers a mass rebuild.

**Comment by jian-lin** [comment]
FYI, I proposed a fix for this: https://github.com/NixOS/nixpkgs/pull/291416

**Comment by Nebucatnetzer** [comment]
> It became a bit too hard recently, since Emacs triggers a mass rebuild.

Ah okay, makes sense.

**Comment by Nebucatnetzer** [comment]
> FYI, I proposed a fix for this: #291416

If I understood this correctly we need this to have it working in 24.05?
Going to test it then.

**Comment by jian-lin** [comment]
Fixed by https://github.com/NixOS/nixpkgs/pull/291416 and https://github.com/NixOS/nixpkgs/pull/292830.

Not sure why this issue is not closed by them though.
